chain-core/src/init/address.rs

```diff
@@ -135,6 +145,25 @@ impl RedeemAddress {
     }
 }
 
+impl CroAddress<RedeemAddress> for RedeemAddress {
+    fn to_cro(&self) -> Result<String, ()> {
+        let checked_data: Vec<u5> = self.0.to_vec().to_base32();
+        Bech32::new("crms".into(), checked_data)
```

Comment:
this hardcodes the network -- perhaps `to_cro` should take the network as a parameter? could be an enum:
```
pub enum Network {
   Mainnet,
   Testnet,
}
```

Reply:
Or maybe perform conditional compilation based on compile time arguments?

Reply:
because chaing network is very frequent, let's go with runtime options.

Reply:
done,


used conditional compilation.
chain-core = { path = "../chain-core" , features = ["mainnet"]}
chain-core = { path = "../chain-core" , features = ["testnet"]}

otherwise, rust don't like mutable global state.
1.
so need to hand over globla state to all classes
2. 
or use unstable block, to access some global state,

both are possible, but need some discussion.
so let's proceed with conditional compilation for now.